src/generate_features.py

```diff
@@ -1,3 +1,215 @@
 import logging
+import os
+from multiprocessing import Pool
+import sys
+import boto3
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import yaml
+from sklearn import preprocessing
+from sklearn.cluster import KMeans
+from sklearn.metrics import calinski_harabaz_score as score_ch
+from sklearn.mixture import GaussianMixture as GMM
 
+
+logging.basicConfig(format="%(asctime)s - %(name)s - %(levelname)s - %(message)s")
 logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+
+ROOT = os.getcwd()
+logger.debug('Project Root = %s', ROOT)
+
+with open(os.path.join(ROOT, 'config', 'features_config.yml'), 'r') as f:
+    config = yaml.load(f)
+with open(os.path.join(ROOT, 'data', 'external', 'cats.yml'), 'r') as f:
```

Comment:
can an external hardcoded list for category lists get confusing? maybe a simple comment above these to explain why two config files were needed

Reply:
cats is not really a config so much as another data file which just happens to be yaml format for simplicity. The previously mentioned documentation for the data dir as well as some inline comments will help clarify